x-pack/plugins/apm/server/routes/infrastructure/route.ts

```diff
@@ -29,12 +30,9 @@ const infrastructureRoute = createApmServerRoute({
     podNames: string[];
   }> => {
     const setup = await setupRequest(resources);
+    const setupInfraMetrics = await setupInfraMetricsRequest(resources);
```

Comment:
Why are we not calling `createInfraMetricsClient` directly?

```suggestion
    const setupInfraMetrics = await createInfraMetricsClient(resources);
```